src/analyze/analyze-time-data.c

```diff
@@ -274,9 +299,11 @@ int acquire_time_data(sd_bus *bus, bool require_finished, UnitTimes **out) {
                                 &error,
                                 NULL,
                                 t);
-                if (r < 0)
+                if (r < 0) {
+                        related_units_free(&t->related);
```

Comment:
I don't think there's any way of getting compiler help at not forgetting these frees here because the related field is not a pointer (and cannot be, due to the reuse of the same d-bus call used to gather other information, which I don't think can be avoided without writing a very ugly mapper function. Possible but it results in arguably more non-localized and ugly code).

Reply:
Since you've embedded the struct, please simply introduce destructor for `UnitTimes`, namely `unit_times_done`, and do all the cleaning there.